pkg/cover/backend/dwarf.go

```diff
@@ -351,7 +351,36 @@ type pcFixFn = (func([2]uint64) ([2]uint64, bool))
 func readTextRanges(debugInfo *dwarf.Data, module *vminfo.KernelModule, pcFix pcFixFn) (
 	[]pcRange, []*CompileUnit, error) {
 	var ranges []pcRange
-	var units []*CompileUnit
+	unitMap := map[string]*CompileUnit{}
+	addRange := func(r [2]uint64, unitName string) {
+		if pcFix != nil {
+			var filtered bool
+			r, filtered = pcFix(r)
+			if filtered {
+				return
+			}
+		}
+		// Rust compilation units have a special naming scheme.
+		if strings.Contains(unitName, "/@/") {
```

Comment:
Move this to rustRanges.

Reply:
I've dropped it completely since for Rust we take file names from the per-line info and don't consider the compilation unit name.